engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -994,33 +980,29 @@ public Answer easySend(final Long hostId, final Command cmd) {
     }
 
     @Override
-    public boolean reconnect(final long hostId) {
-        HostVO host;
+    public void reconnect(final long hostId) throws AgentUnavailableException {
+        HostVO host = _hostDao.findById(hostId);
+        if (host == null) {
+            throw new CloudRuntimeException("Unable to find host: " + hostId);
+        }
 
-        host = _hostDao.findById(hostId);
-        if (host == null || host.getRemoved() != null) {
-            s_logger.warn("Unable to find host " + hostId);
-            return false;
+        if (host.getRemoved() != null) {
+            throw new CloudRuntimeException("Host has already been removed: " + hostId);
         }
 
         if (host.getStatus() == Status.Disconnected) {
-            s_logger.info("Host is already disconnected, no work to be done");
-            return true;
+            s_logger.debug("Host is already disconnected, no work to be done: " + hostId);
```

Comment:
@DaanHoogland I made a mistake here. Can you fix it for me?
You need to add a `return;`, right after line 994